src/frontend/src/handler/alter_table_with_sr.rs

```diff
@@ -81,14 +75,34 @@ pub async fn handle_refresh_schema(
         .try_into()
         .unwrap();
 
-    replace_table_with_definition(
+    let result = replace_table_with_definition(
         &session,
         table_name,
         definition,
         &original_table,
         Some(connector_schema),
     )
-    .await?;
+    .await;
 
-    Ok(RwPgResponse::empty_result(StatementType::ALTER_TABLE))
+    match result {
+        Ok(_) => Ok(RwPgResponse::empty_result(StatementType::ALTER_TABLE)),
+        Err(e) => {
+            let report = e.to_report_string();
+            // This is a workaround for reporting errors when columns to drop is referenced by generated column.
+            // Finding the actual columns to drop requires generating `PbSource` from the sql definition
+            // and fetching schema from schema registry, which will cause a lot of unnecessary refactor.
+            // The only cause of those matched errors is the generated column referencing the column to drop.
+            if report.contains("Failed to bind expression")
+                && report.contains("Item not found: Invalid column")
```

Comment:
Hope you like it @BugenZhao

Reply:
It's obscure and unintuitive that "failed to bind expression" is specific to binding the expression for generated columns here. I would rather suggest adding a context like "failed to bind generated column" below, then exactly match that prompt.

https://github.com/risingwavelabs/risingwave/blob/3142219c0411713cf37ff1cdacf2c8ee802674da/src/frontend/src/handler/create_table.rs#L291